src/controllers/searchResult/base.js

```diff
@@ -54,12 +55,14 @@ class SearchResultBase {
         hit = this._response.hits && this._response.hits[this._response.hits.length -1];
 
       for (const sort of this._request.sort) {
-        if (typeof sort === 'string') {
-          request.search_after.push(hit._source[sort]);
-        }
-        else {
-          request.search_after.push(hit._source[Object.keys(sort)[0]]);
-        }
+        const key = typeof sort === 'string'
+          ? sort
+          : Object.keys(sort)[0];
+        const value = key === '_uid'
+          ? hit._id
```

Comment:
according to ES documentation, I understand that if results are sorted by `_uid`, the "search_after" item whould be `this._request.collection + '#" + hit._id` instead (see https://www.elastic.co/guide/en/elasticsearch/reference/5.6/search-request-search-after.html )
and we also should consider sorting by `_id`

Reply:
:+1:  good catch